packages/@aws-cdk/aws-cloudfront/lib/public-key.ts

```diff
@@ -0,0 +1,81 @@
+import { Names, Resource, Token } from "@aws-cdk/core";
+import { Construct } from "constructs";
+import { CfnPublicKey } from "./cloudfront.generated";
+
+/**
+ * Represents a Public Key
+ */
+export interface IPublicKey {
+  /**
+   * The ID of the key group.
+   * @attribute
+   */
+  readonly publicKeyId: string;
+}
+
+/**
+ * Properties for creating a Public Key
+ */
+export interface PublicKeyProps {
+  /**
+   * A name to identify the key group.
+   * @default - generated from the `id`
+   */
+  readonly publicKeyName?: string;
+
+  /**
+   * A comment to describe the public key.
+   * @default - no comment
+   */
+  readonly comment?: string;
+
+  /**
+   * A string included in the request to help make sure that the request can’t be replayed.
+   */
+  readonly callerReference: string;
```

Comment:
I suspect we can default for this as well; I almost wonder if we should just (at least initially) omit it from the props, as this kind of request-time nonce doesn't mesh well with the mental model of infrastructure-as-code and the CDK. I can't imagine users explicitly setting this value on each deploy to something new. What do you think?

Reply:
I totally agree. I assume since we'll omit it I'll have to add `awslint` exclusions in the package.json for this property, right?

Reply:
No. Only attributes and other specific properties (e.g., `fooName`) are expected and checked by `awslint`. Others like this can be freely omitted without complaint.

Reply:
👍

Reply:
@njlynch CfnPublicKey constructor requires `callerReference` to be set. Should I just set it to an empty string to satisfy the requirement?

Reply:
How about something non-null and somewhat unique? I think something like `this.node.addr` would be a good candidate.